sdk/core/azure-core/src/main/java/com/azure/core/util/polling/PollingStrategy.java

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.util.polling;
+
+import com.azure.core.http.rest.Response;
+import com.azure.core.util.BinaryData;
+import com.azure.core.util.serializer.TypeReference;
+import reactor.core.publisher.Mono;
+
+/**
+ * Represents a known strategy for polling a long running operation in Azure.
+ *
+ * <p>
+ *
+ * The methods in the polling strategy will be invoked from the {@link com.azure.core.util.polling.PollerFlux}. The
+ * order of the invocations is:
+ *
+ * <ol>
+ *     <li>{@link #canPoll(Response)} - exits if returns false</li>
+ *     <li>{@link #onInitialResponse(Response, PollingContext, TypeReference)} - immediately after
+ *          {@link #canPoll(Response)} returns true</li>
+ *     <li>{@link #poll(PollingContext, TypeReference)} - invoked after each polling interval, if the last polling
+ *          response indicates an "In Progress" status. Returns a {@link PollResponse} with the latest status</li>
+ *     <li>{@link #getResult(PollingContext, TypeReference)} - invoked when the last polling response indicates a
+ *          "Successfully Completed" status. Returns the final result of the given type</li>
+ * </ol>
```

Comment:
Do we expect poller flux to call any of these methods concurrently or is it always in a series?

Reply:
Always in series.